python-client/giskard/integrations/mlflow/giskard_evaluator.py

```diff
@@ -0,0 +1,101 @@
+
+
+class GiskardEvaluator(ModelEvaluator):
+    def can_evaluate(self, model_type, **kwargs):
+        return model_type in ["classifier", "regressor"]
+
+    def evaluate(
+            self, *, model, model_type, dataset, run_id, evaluator_config, baseline_model=None, **kwargs
+    ):
+        self.client = MlflowClient()
+        self.run_id = run_id
+        self.evaluator_config = evaluator_config
+
+        if not isinstance(dataset.features_data, pd.DataFrame):
+            raise ValueError("Only pd.DataFrame are currently supported by the giskard evaluator.")
+        else:
+            try:
+                data = dataset.features_data.copy()
+                data[dataset.targets_name] = dataset.labels_data
+                giskard_dataset = Dataset(df=data,
+                                          target=dataset.targets_name,
+                                          name=dataset.name)
+
+                # log dataset
+                giskard_dataset.to_mlflow(client=self.client, run_id=self.run_id)
+            except Exception as e:
+                raise ValueError("An error occurred while wrapping the dataset. "
+                                 "Please submit the traceback as a GitHub issue in the following "
+                                 "repository for further assistance: https://github.com/Giskard-AI/giskard.") from e
+
+        try:
+            giskard_model = PyFuncModel(model=model,
+                                        model_type=gsk_model_types[model_type],
+                                        feature_names=dataset.feature_names,
+                                        **evaluator_config)
+        except Exception as e:
+            raise ValueError("An error occurred while wrapping the model. "
+                             "Please submit the traceback as a GitHub issue in the following "
+                             "repository for further assistance: https://github.com/Giskard-AI/giskard.") from e
+
+        try:
+            validation_flags = ValidationFlags()
+            validation_flags.model_loading_and_saving = False
+            scan_results = scan(model=giskard_model, dataset=giskard_dataset, validation_flags=validation_flags)
+
+            # log html scan result
+            scan_results.to_mlflow(client=self.client, run_id=self.run_id)
+        except Exception as e:
+            raise ValueError("An error occurred while scanning the model for vulnerabilities. "
+                             "Please submit the traceback as a GitHub issue in the following "
+                             "repository for further assistance: https://github.com/Giskard-AI/giskard.") from e
+
+        try:
+            test_suite = scan_results.generate_test_suite("scan test suite")
+            test_suite_results = test_suite.run()
+
+            # log metrics resulting from scan
+            test_suite_results.to_mlflow(client=self.client, run_id=self.run_id)
+        except Exception:  # noqa
+            pass
```

Comment:
we should at least send a telemetry report in this case. Also why not log this error?

Reply:
Even if the test suite wasn't generated, the user can still the results of the scan and the other artifacts. I will issue a warning instead. And yes I added telemetry.

Reply:
the telemetry is minimal, since information is already collected in the scan. But I picked up the dataset and model ids under the mlflow-integration so we can identify when people use giskard through the mlflow plugin. Is that enough?

How can we make it simple from mixpanel to separate people that go through scan from mlflow-plugin and not explicitely? if there's more things to log to make this happen, let me know

Reply:
in fact due to https://linear.app/giskard/issue/GSK-1328/symmetrise-dataset-and-model-upload-process-and-id-creation, i had to save the model and inner model names instead of ids, since I don't validate the saving of the model during scan, the model id is not created. We should fix this.